### PR TITLE
add readme explaining how packet.net provisioning script works

### PR DESCRIPTION
**Pull request by paulfantom**
SSIA

/assign @ingvagabund

### CODE REVIEW

File: hack/README.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
+# Packet instance provisioning
+
+Our CI system uses packet.net to provision instances and tests code there. To accomplish this we are running 
+`packet-provision.sh` script which is a thin wrapper on terraform code. This mechanism creates machines based on images
+build by using [paulfantom/packet-image](https://github.com/paulfantom/packet-image) repository.
```
Comment (frobware) [inline hack/README.md]: typo: "built by"

File: hack/README.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+Script is able to provision and deprovision one instance in packet.net. It also takes care of tagging that instance,
+assigning unique name to it and creating temporary ssh key (Private key is located in `/tmp/packet_id_rsa` unless 
+`TF_VAR_ssh_key_path` was previously set).
+Script can be configured with following environment variables:
+- `PACKET_AUTH_TOKEN` - packet.net API authentication token.
+- `TF_VAR_packet_project_id` - packet.net project id.
+- `TF_VAR_ssh_key_path` - path to SSH key (optional)
+
+Script accepts on of two parameters:
```
Comment (frobware) [inline hack/README.md]: type: "one of"

File: hack/README.md [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+- `install` - provision instance
+- `destroy` - cleanup
+
+##### Image reference
+
+Currently we bake in reference to image which we want to use by using a commit id from master branch of
```
Comment (frobware) [inline hack/README.md]: typo: "we bake in a" ... "to the image".

File: hack/README.md [hunk 4]
```diff
@@ -0,0 +1,25 @@
+# Packet instance provisioning
+
+Our CI system uses packet.net to provision instances and tests code there. To accomplish this we are running
```
Comment (ingvagabund) [inline hack/README.md]: `s/tests/to test/`

File: hack/README.md [hunk 5]
```diff
@@ -0,0 +1,25 @@
+# Packet instance provisioning
+
+Our CI system uses packet.net to provision instances and tests code there. To accomplish this we are running 
+`packet-provision.sh` script which is a thin wrapper on terraform code. This mechanism creates machines based on images
```
Comment (ingvagabund) [inline hack/README.md]: `s/on/over/`

File: hack/README.md [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+build by using [paulfantom/packet-image](https://github.com/paulfantom/packet-image) repository.
+
+##### packet-provision.sh
+
+Script is able to provision and deprovision one instance in packet.net. It also takes care of tagging that instance,
+assigning unique name to it and creating temporary ssh key (Private key is located in `/tmp/packet_id_rsa` unless
```
Comment (ingvagabund) [inline hack/README.md]: `s/Private/private/`

File: hack/README.md [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+`packet-provision.sh` script which is a thin wrapper on terraform code. This mechanism creates machines based on images
+build by using [paulfantom/packet-image](https://github.com/paulfantom/packet-image) repository.
+
+##### packet-provision.sh
+
+Script is able to provision and deprovision one instance in packet.net. It also takes care of tagging that instance,
+assigning unique name to it and creating temporary ssh key (Private key is located in `/tmp/packet_id_rsa` unless 
+`TF_VAR_ssh_key_path` was previously set).
+Script can be configured with following environment variables:
+- `PACKET_AUTH_TOKEN` - packet.net API authentication token.
```
Comment (ingvagabund) [inline hack/README.md]: Drop the ending `.`. Does not make much sense unless the description starts with capital letter.

File: hack/README.md [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+- `TF_VAR_packet_project_id` - packet.net project id.
```
Comment (ingvagabund) [inline hack/README.md]: Drop the ending `.`. Does not make much sense unless the description starts with capital letter.

File: hack/README.md [hunk 9]
```diff
@@ -0,0 +1,25 @@
...
+
+Script accepts on of two parameters:
+- `install` - provision instance
+- `destroy` - cleanup
+
+##### Image reference
+
+Currently we bake in reference to image which we want to use by using a commit id from master branch of
+[paulfantom/packet-image](https://github.com/paulfantom/packet-image) repository. This is done in: 
+https://github.com/openshift/cluster-api-provider-libvirt/blob/master/hack/prebuild/main.tf#L14
```
Comment (ingvagabund) [inline hack/README.md]: That line can move in time. Can you instead use relative path? E.g. `user_data` field of `resource "packet_device" "libvirt"` at LINK.

File: hack/README.md [hunk 10]
```diff
@@ -0,0 +1,25 @@
...
+`TF_VAR_ssh_key_path` was previously set).
+Script can be configured with following environment variables:
+- `PACKET_AUTH_TOKEN` - packet.net API authentication token.
+- `TF_VAR_packet_project_id` - packet.net project id.
+- `TF_VAR_ssh_key_path` - path to SSH key (optional)
+
+Script accepts on of two parameters:
+- `install` - provision instance
+- `destroy` - cleanup
+
```
Comment (ingvagabund) [inline hack/README.md]: Maybe put example here?
```sh
PACKET_AUTH_TOKEN="..." ./hack/packet-provision.sh install
```

**Comment by ingvagabund** [comment]
Also worth mentioning invoker of the script is responsible for calling `destroy` command so all provisioned resource are cleaned at the end to avoid exhaustion.

**Comment by paulfantom** [comment]
@ingvagabund were you changing sth in jenkins pipeline?

/retest

**Pull request merged**